src/procfs.rs

```diff
@@ -252,6 +252,34 @@ pub fn get_node_information(system: &dyn systemapi::SystemAPI) -> Result<(u64, V
     Ok((cpu_total_secs, per_cpu_secs))
 }
 
+pub fn get_loadavg(fs: &dyn procfsapi::ProcfsAPI) -> Result<(f64, f64, f64, u64, u64), String> {
+    let s = fs.read_to_string("loadavg")?;
+    let fields = s.split_ascii_whitespace().collect::<Vec<&str>>();
+    if fields.len() != 5 {
+        return Err("Unexpected loadavg".to_string());
+    }
+    let load1 = fields[0]
+        .parse::<f64>()
+        .map_err(|_| "Bad loadavg".to_string())?;
+    let load5 = fields[1]
+        .parse::<f64>()
+        .map_err(|_| "Bad loadavg".to_string())?;
+    let load15 = fields[2]
+        .parse::<f64>()
+        .map_err(|_| "Bad loadavg".to_string())?;
+    let entities = fields[3].split('/').collect::<Vec<&str>>();
+    if entities.len() != 2 {
+        return Err("Unexpected entities".to_string());
+    }
+    let runnable = entities[0]
+        .parse::<u64>()
+        .map_err(|_| "Bad entity".to_string())?;
+    let existing = entities[1]
+        .parse::<u64>()
+        .map_err(|_| "Bad entity".to_string())?;
```

Comment:
The error messages are not good - the last two are worse than the others, too.  Basically these will never happen, but if they do happen we want maximal information, so: better prose, and include the input line in the messages.